cpp/src/slice2java/Gen.cpp

```diff
@@ -4366,11 +4374,26 @@ Slice::Gen::ProxyVisitor::visitInterfaceDefEnd(const InterfaceDefPtr& p)
     {
         outi << nl << "@Deprecated";
     }
-    outi << nl << "public class _" << p->name() << "PrxI";
-    outi << " extends com.zeroc.Ice._ObjectPrxFactoryMethods<" << p->name() << "Prx>";
-    outi << " implements " << p->name() << "Prx";
+    outi << nl << "public class " << prxIName;
+    outi << " extends com.zeroc.Ice._ObjectPrxFactoryMethods<" << prxName << ">";
+    outi << " implements " << prxName;
     outi << sb;
 
+    // TODO: eventually remove this default constructor.
+    // Default constructor
+    outi << sp;
+    outi << nl << "public " << prxIName << "()";
+    outi << sb;
+    outi << "super();";
+    outi << eb;
+
+    // Copy constructor
+    outi << sp;
+    outi << nl << "public " << prxIName << "(com.zeroc.Ice.ObjectPrx obj)";
```

Comment:
Can you make this constructor private, like in C#?
https://github.com/zeroc-ice/ice/blob/b81c856d81add8691dafe56ac51097faed13a9da/cpp/src/slice2cs/Gen.cpp#L3254

Reply:
It's not possible with the current structure, because `checkedCast` is implemented on `FooPrx`, and these constructors are for `_FooPrxI`. I made them package private instead, because I think that's the best we can do here.

Reply:
So you're making it package private?

Reply:
Yes, the generated constructors are package private,
the ProxyFactory method class's constructor is protected,
and the base ObjectPrx constructor is public.

Reply:
Nevermind, the generated constructors also need to be public.
There are still some places where we're using reflection to construct proxies, and the reflection that we're doing only works with public items.

I'd rather mark them as public (for now), then fix our reflection code to work with non-public items.
This will all be going away in the future anyways, and then we can fix the visibility later.